TRRandomizerCore/Meshes/MeshEditor.cs

```diff
@@ -203,6 +203,40 @@ public TRMesh CloneMesh(TRMesh mesh)
             return clone;
         }
 
+        public TRMesh CloneMeshAsColoured(TRMesh mesh, ushort paletteIndex)
+        {
+            TRMesh clone = CloneMesh(mesh);
+
+            clone.ColouredRectangles = new TRFace4[mesh.NumTexturedRectangles];
+            clone.ColouredTriangles = new TRFace3[mesh.NumTexturedTriangles];
+            clone.NumColouredRectangles = mesh.NumTexturedRectangles;
+            clone.NumColouredTriangles = mesh.NumTexturedTriangles;
+            clone.NumTexturedRectangles = 0;
+            clone.NumTexturedTriangles = 0;
+            clone.TexturedRectangles = new TRFace4[] { };
+            clone.TexturedTriangles = new TRFace3[] { };
+
+            for (int i = 0; i < mesh.NumTexturedRectangles; i++)
+            {
+                clone.ColouredRectangles[i] = new TRFace4
+                {
+                    Texture = paletteIndex,
+                    Vertices = mesh.TexturedRectangles[i].Vertices
```

Comment:
Do the Vertices copies need to be deep clones? The clone and the original are sharing vertices as it is. Same for the triangles.

Reply:
Good point. I've fixed this now, and also in the other clone method.